web/src/app/util/colors.ts

```diff
@@ -0,0 +1,166 @@
+import { round } from 'lodash-es'
+import seedrandom from 'seedrandom'
+
+export type Color = [number, number, number]
+
+// getAllyColors generates a set of n random colors with a
+// contrast ratio of at least 4.5:1 against a white background
+// as per WCAG standards
+export function getAllyColors(seed: string, num = 1): Color[] {
+  const seedRng = seedrandom(seed)
+
+  let colors = []
+  for (let i = 0; i < num; i++) {
+    const colorSeed = seedrandom((seedRng() + i).toString())
+
+    // every color needs an red, green, and blue value from 0-255
+    let rgb: Color = [0, 0, 0]
+    for (let j = 0; j < 3; j++) {
+      const rng = seedrandom((colorSeed() + j).toString())()
+      rgb[j] = Math.floor(((rng + 1) * 255) / 2)
+    }
+
+    colors.push(makeColorA11y(rgb))
+  }
+
+  return colors
+}
```

Comment:
I'm thinking we should keep the color generation as a script/tool but for this purpose we should define a fixed-pallet of pre-defined colors that are all different enough from the background, as well as from each other.

The reason is that given the same input, we need the exact same sequence of colors as an output for this and all future versions of GoAlert. Phone apps will update at a different schedule than individual GoAlert instances, and a single phone could talk to multiple GoAlert versions.

Changing this function, or any of its dependencies could cause a change in the output, and there would then be a discrepancy between devices. To avoid that we can generate a set of colors one time (maybe 8 or something) and then select them for a given code.

Reply:
This is fair, I'd definitely like to keep the tool as I have plans to expand upon it as a palette generator for further use in GoAlert for different themes, while staying a11y compatible on our terms (no 3rd party library dependency). Perhaps I can pull this tool out since it's in a different file with @jakegleason's help on the unit tests and put it in a separate draft PR that we can work on for a **_Colors_** milestone. That would be fun 🙂 

---

Are you thinking then, with a little bit toned down on the randomness, that we cycle through a set of 8 colors or something? If we did a pick on an array of pre-determined colors, there's a good chance that the colors could be the same if the codes are different.

Perhaps this is all overkill, and a simple typography display of the code is enough, since colors as a concept aren't a11y in the first place because of colorblindness. 🤷‍♀️

Lemme know your thoughts @mastercactapus, would love to hear them

Reply:
The colors aren't just for readability -- they are meant as an additional hint/indicator that the two codes are different.

If all colors end up being the same, but if a single character of the code changes the sequence of colors would be very different, even if two codes are a single letter apart.

Basically, the user is matching codes and/or a sequence of colors. `ABCDEF` is very similar to `ABDCEF` at first glance, but the two would have wildly different color sequences which would help prevent a user from glancing and assuming they are the same.

Our pallet for this would be fixed and shared between web and mobile (if originally generated randomly) and:
- all colors should have an acceptable contrast from all other colors in the list
- all colors should have an acceptable contrast from a white background
- all colors should have an acceptable contrast from a dark background (i.e. respect dark mode in a browser and/or mobile)
- should _probably_ have at least the number of colors match the number of characters we're thinking for the code